modules/apps/headless/headless-admin-taxonomy/headless-admin-taxonomy-impl/src/main/java/com/liferay/headless/admin/taxonomy/internal/dto/v1_0/converter/TaxonomyCategoryDTOConverter.java

```diff
@@ -172,34 +173,37 @@ private TaxonomyCategory _toTaxonomyCategory(
 						};
 					});
 				setTaxonomyCategoryUsageCount(
-					() -> {
-						UriInfo uriInfo = dtoConverterContext.getUriInfo();
-
-						if (uriInfo != null) {
-							MultivaluedMap<String, String> queryParameters =
-								uriInfo.getQueryParameters();
-
-							if (StringUtil.contains(
-									queryParameters.getFirst("restrictFields"),
-									"taxonomyCategoryUsageCount")) {
-
-								return null;
+					NestedFieldsSupplier.<Integer>supply(
+						"taxonomyCategoryUsageCount",
+						fieldName -> {
+							UriInfo uriInfo = dtoConverterContext.getUriInfo();
+
+							if (uriInfo != null) {
+								MultivaluedMap<String, String> queryParameters =
+									uriInfo.getQueryParameters();
+
+								if (StringUtil.contains(
+										queryParameters.getFirst(
+											"restrictFields"),
```

Comment:
Maybe it would be a great idea to filter the nested fields if the `restrictFields` parameter contains a required nested field